jupyterhub/tests/test_auth_expiry.py

```diff
@@ -101,6 +101,7 @@ async def test_auth_expired_page(app, user, disable_refresh):
     assert user._auth_refreshed == before
 
 
+# Fixme: Why does this text fail?
```

Comment:
This is failing because `BaseHandler.refresh_user()` called in `get_current_user()` refreshes auth and can take a resolved user back to None.

I think the fix is that `_resolve_scopes` should return an empty set if `get_current_user()` is None, even if `get_token()` returns a token.